internal/cli/atlas/deployments/diagnostics.go

```diff
@@ -34,72 +33,67 @@ type diagnosticsOpts struct {
 	cli.GlobalOpts
 	options.DeploymentOpts
 	podmanClient podman.Client
-	podmanDiag   *podman.Diagnostic
-	machineDiag  *machineDiagnostic
-	podmanLogs   map[string]interface{}
-	mongotLogs   []string
-	mongodLogs   []string
 }
 
+type diagnosticLogs struct {
+	MongoD []string
+	MongoT []string
+}
+type diagnostic struct {
+	Machine    machineDiagnostic
+	Diagnostic *podman.Diagnostic
+	Containers []*define.InspectContainerData
+	Logs       diagnosticLogs
+	Network    *types.Network
+	Errors     []error
+}
 type machineDiagnostic struct {
 	OS   string
 	Arch string
 }
 
 func (opts *diagnosticsOpts) Run(ctx context.Context) error {
-	var err error
-
-	// Machine system info
-	opts.machineDiag = &machineDiagnostic{
-		OS:   runtime.GOOS,
-		Arch: runtime.GOARCH,
+	d := &diagnostic{
+		Machine: machineDiagnostic{
+			OS:   runtime.GOOS,
+			Arch: runtime.GOARCH,
+		},
+		Diagnostic: opts.podmanClient.Diagnostics(ctx),
 	}
 
-	// Podman system info
-	opts.podmanDiag = opts.podmanClient.Diagnostics(ctx)
-
-	// Podman logs
-	var errs []error
-	if opts.podmanLogs, errs = opts.podmanClient.Logs(ctx); errs != nil {
-		for _, e := range errs {
-			opts.podmanDiag.Errors = append(opts.podmanDiag.Errors, fmt.Errorf("failed to get podman logs: %w", e).Error())
-		}
+	var err error
+	d.Containers, err = opts.podmanClient.ContainerInspect(ctx, opts.LocalMongodHostname(), opts.LocalMongotHostname())
+	if err != nil {
+		d.Errors = append(d.Errors, err)
 	}
 
-	if opts.DeploymentName != "" {
-		_, _ = log.Warningf("Fetching logs for deployment %s\n", opts.DeploymentName)
-		// ignore error if container does not exist just capture log for that command
-		if opts.mongotLogs, err = opts.podmanClient.ContainerLogs(ctx, opts.LocalMongotHostname()); err != nil {
-			opts.podmanDiag.Errors = append(opts.podmanDiag.Errors, fmt.Errorf("failed to get mongot logs: %w", err).Error())
-		}
-		if opts.mongodLogs, err = opts.podmanClient.ContainerLogs(ctx, opts.LocalMongodHostname()); err != nil {
-			opts.podmanDiag.Errors = append(opts.podmanDiag.Errors, fmt.Errorf("failed to get mongod logs: %w", err).Error())
-		}
+	n, nErr := opts.podmanClient.Network(ctx, opts.LocalNetworkName())
+	if nErr != nil {
+		d.Errors = append(d.Errors, nErr)
 	}
+	d.Network = n[0]
```

Comment:
if the provided name is invalid, this will throw an nil panic, can w maybe only assign if there is a value?